File: FirClient/Assets/Scripts/Data/TableManager.cs
namespace FirCommon.Data
{
	public class TableManager : BaseObject
	{
		private static TableManager instance;
    	public NpcTable npcTable;
    	public ObjectPoolTable objectPoolTable;
    	public GlobalConfigTable globalConfigTable;
///[APPEND_VAR]

		public static TableManager Create()
		{
			if (instance == null)
			{
				instance = new TableManager();
			}
			return instance;
		}

		public override void Initialize()
		{
			this.LoadTables();
		}
			
		public T LoadData<T>(string path) where T : class
		{
            var fullPath = base.DataPath + path;
            return FirCommon.Utility.ProtoUtil.Deserialize<T>(fullPath);
		}

		public void LoadTables() 
		{
        	npcTable = LoadData<NpcTable>("Tables/NpcTable.bytes");
        	npcTable.Initialize();
        	objectPoolTable = LoadData<ObjectPoolTable>("Tables/ObjectPoolTable.bytes");
        	objectPoolTable.Initialize();
        	globalConfigTable = LoadData<GlobalConfigTable>("Tables/GlobalConfigTable.bytes");
        	globalConfigTable.Initialize();
///[APPEND_TABLE]
		}

		public override void OnUpdate(float deltaTime)
		{
		}

		public override void OnDispose()
		{
		}
	}
}

File: FirClient/Assets/Scripts/Lua/Data/TableManager.lua
local TableManager = class("TableManager")

function TableManager:__index(k)
    if TableManager.tableMgr == nil then
        TableManager.tableMgr = MgrCenter:GetExtManager("TableManager")
    end
	return TableManager.tableMgr[k]
end

function TableManager:Initialize()
    TableManager.itemTable = require 'Data.Tables.ItemTable'
    TableManager.itemTable:Initialize()
    TableManager.qualityTable = require 'Data.Tables.QualityTable'
    TableManager.qualityTable:Initialize()
---[APPEND_VAR]---
end

return TableManager